docs/insert_sql_userguide.md

```diff
@@ -0,0 +1,127 @@
+
+## Use Spark SQL
+1. Initiate `SparkConf`.
+
+    ```scala
+    val sparkConf = new SparkConf().
+      setIfMissing("spark.master", "local[*]").
+      setIfMissing("spark.app.name", getClass.getName).
+      setIfMissing("spark.sql.extensions", "org.apache.spark.sql.TiExtensions").
+      setIfMissing("spark.tispark.pd.addresses", "pd0:2379").
+      setIfMissing("spark.tispark.tidb.addr", "tidb").
+      setIfMissing("spark.tispark.tidb.port", "4000")
+
+    val spark = SparkSession.builder.config(sparkConf).getOrCreate()
+    ```
+
+2. Write using Spark SQL.
+
+    ```scala
+    /* create table before run the code
+    CREATE TABLE tpch_test.target_table_orders (
+      `O_ORDERKEY` int(11) NOT NULL,
+      `O_CUSTKEY` int(11) NOT NULL,
+      `O_ORDERSTATUS` char(1) NOT NULL,
+      `O_TOTALPRICE` decimal(15,2) NOT NULL,
+      `O_ORDERDATE` date NOT NULL,
+      `O_ORDERPRIORITY` char(15) NOT NULL,
+      `O_CLERK` char(15) NOT NULL,
+      `O_SHIPPRIORITY` int(11) NOT NULL,
+      `O_COMMENT` varchar(79) NOT NULL
+    )
+    */
+    
+   // write data to tidb
+    spark.sql(
+          "insert into tpch_test.target_table_orders values" +
+            "(1,1,'0',11.11,date'2022-1-1','first','spark',12345,'no comment')")
+
+    // select data to write
+    val df = spark.sql("select * from tpch_test.ORDERS")
+   ```
+
+## TiDB Options
+
+The following table shows the TiDB-specific options, which can be passed in through `SparkConf`.
+
+| Key                         | Required | Default         | Description                                                                                                                   |
+|-----------------------------|----------|-----------------|-------------------------------------------------------------------------------------------------------------------------------|
+| tidb.addr                   | false    | -               | TiDB address, needed when enableUpdateTableStatistics is true                                                                 |
+| tidb.port                   | false    | -               | TiDB Port, needed when enableUpdateTableStatistics is true                                                                    |
+| tidb.user                   | false    | -               | TiDB User, needed when enableUpdateTableStatistics is true                                                                    |
+| tidb.password               | false    | -               | TiDB Password, needed when enableUpdateTableStatistics is true                                                                |
+| enableRegionSplit           | false    | true            | To split Region to avoid hot Region during insertion                                                                          |
+| scatterWaitMS               | false    | 300000          | Max time to wait scatter region                                                                                               |
```

Comment:
Why delete doesn't have so many options?

Reply:
This options are used in write. Some of them are used to create jdbc client to update statistics, some of them are used to do region splits. 

Delete doesn't do these things. So it doesn't need.